### PR TITLE
Sdk 267 tps python tests

### PR DESCRIPTION
**Pull request by rushby**
Performance test framework with 4 different test types available.  This is for merging into the tps-tests branch where additional functionality can be added.

### CODE REVIEW

File: qa/perf_test.py [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+        return result["block"]["sidechainTransactions"]
+
+    def txs_creator_send_to_address(self, utxo_amount, txs_creator_node, non_creator_nodes, tps_test):
+        node_index = self.sc_nodes.index(txs_creator_node)
+        destination_address = non_creator_nodes[0].wallet_createPrivateKey25519()["result"]["proposition"][
+            "publicKey"]
+        start_time = time.time()
+
+        if tps_test:
+            max_tps_per_process = 50
```
Comment (MarcoOl94) [inline qa/perf_test.py:235]: Please add some comments to explain these.

**Review by MarcoOl94 — CHANGES_REQUESTED** [review]

File: qa/performance/perf_data.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    initial_txs: int
+    network_topology: int
+    nodes: list[NodeData]
+
+
+class NodeData(TypedDict, total=False):
+    latency_settings: int
+    forger: bool
+    tx_creator: bool
+    throughput: int
```
[the remark this answers is not in the document]
Comment (rushby) [reply qa/performance/perf_data.py]: throughput and latency is removed, the other 2 are referenced as part of the PerformanceData object

File: qa/performance/perf_test.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
```
[the remark this answers is not in the document]
Comment (rushby) [reply qa/performance/perf_test.json]: Done

File: qa/perf_test.py [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+                if node == 0:
+                    node += 1
+                connect_sc_nodes(self.sc_nodes[node], 0)
+                self.create_node_connection_map(0, [node])
+                self.create_node_connection_map(node, [0])
+                node += 1
+        print(f"NETWORK TOPOLOGY CONNECTION MAP: {self.connection_map}")
+        self.sc_sync_all()
+
+    def find_txs_creators(self):
```
[the remark this answers is not in the document]
Comment (rushby) [reply qa/perf_test.py:167]: Changed to:

`get_txs_creators_and_non_creators`

File: qa/perf_test.py [hunk 5]
```diff
@@ -0,0 +1,487 @@
...
+        return txs_creators, non_txs_creator
+
+    def find_forger_nodes(self):
+        forger_nodes = []
+        for index, node in enumerate(self.sc_nodes_list):
+            if node["forger"]:
+                forger_nodes.append(self.sc_nodes[index])
+        return forger_nodes
+
+    def send_coins_to_multiple_destination(self, utxo_amount, txs_creator_nodes, non_creator_nodes):
```
[the remark this answers is not in the document]
Comment (rushby) [reply qa/perf_test.py:184]: Changed to:

`populate_mempool`

File: qa/perf_test.py [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+            ft_addresses.append(txs_creators[i].wallet_createPrivateKey25519()["result"]["proposition"]["publicKey"])
+            forward_transfer_to_sidechain(self.sc_nodes_bootstrap_info.sidechain_id, mc_nodes[0],
+                                          ft_addresses[i], ft_amount, mc_return_address)
+            txs_creator_addresses.append(
+                txs_creators[i].wallet_createPrivateKey25519()["result"]["proposition"]["publicKey"])
+        self.sc_sync_all()
+
+        # Generate 1 SC block to include FTs.
+        forger_nodes = self.find_forger_nodes()
+        # TODO: Cater for multiple forgers
```
[the remark this answers is not in the document]
Comment (rushby) [reply qa/perf_test.py:296]: Removed the TODO comment

File: qa/perf_test.py (hunk 1 above)
Comment (rushby) [reply qa/perf_test.py:235]: Added a lot of comments to explain the code

**Pull request closed without merging**